### PR TITLE
[14.0][MIG] account_consolidation: Migration to 14.0

### PR DESCRIPTION
**Pull request by kaynnan**
cc @marcelsavegnago @etobella @pedrobaeza @augustodinizl @douglascstd 

For conflict resolution purposes, the `account_consolidation` module has been renamed to `account_consolidation_oca`.

### CODE REVIEW

**Comment by kaynnan** [comment]
@etobella @pedrobaeza I believe that the previous conflicts that occurred are now normalized. Thanks for the help in the previous PRs

**Comment by etobella** [comment]
As we are renaimimg it, we should need to add the rename on openupgrade. Do you know how?

**Comment by kaynnan** [comment]
Well, I've seen people do this at OCA, but I don't remember how to do it, what would it be like?

**Comment by etobella** [comment]
no problem, you should need to go to the OpenUpgrade repository and modify the apriori file.

Here https://github.com/OCA/OpenUpgrade/blob/16.0/openupgrade_scripts/apriori.py#L19

You should need to add 

```python
# OCA/account-reconcile
"account_reconcile": "account_reconcile_oca",
```

**Comment by kaynnan** [comment]
Thanks, I made the pull request to change this account_consolidation -> account_consolidation_oca in OpenUpgrade

**Comment by marcelsavegnago** [comment]
@kaynnan I understand that commits must be for migration and not for the inclusion of a new module

**Comment by kaynnan** [comment]
ping @douglascstd @etobella @pedrobaeza  @marcelsavegnago 

Could you review it again, please? The pull request (Script migration) in OpenUpgrade has already been migrated

File: account_consolidation_oca/wizard/consolidation_base.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        return self.env.company
+
+    @api.model
+    def _get_consolidation_profiles(self):
+        return self._default_company().consolidation_profile_ids
+
+    @api.model
+    def default_get(self, fields):
+        """Raise an error if user is not connected to consolidation company."""
+        if not self.env.user.company_id.is_consolidation:
```
Comment (etobella) [inline account_consolidation_oca/wizard/consolidation_base.py]: self.env.company

**Review by etobella — COMMENTED** [review]
I am unable to consolidate again a company. the moves are duplicated

image

check on runbot

http://oca-account-consolidation-14-0-pr41-d9d41f7ea8c1.runboat.odoo-community.org/web#action=196&model=account.move&view_type=list&cids=2%2C1%2C3%2C4&menu_id=101

**Comment by kaynnan** [comment]
> I am unable to consolidate again a company. the moves are duplicated

@etobella thank you for the review. I believe it's correct now, without duplications. I made an adjustment, replacing `self.env.cr.execute` with `self.env['account.move.line'].search()` along with the appropriate search and update rules

**Comment by kaynnan** [comment]
ping

**Comment by etobella** [comment]
Still not working. I am doing the following:

- Create 2 companies  and make one of them consolidated
- Add some movements for previous month on subsidiary
- Match all the accounts (I use the same chart of accounts)
- Consolidate previous month
- A move is created
- Consolidate previous month again
- A move is created (should modify the old one or delete it)

Also, previously the moves were marked to be merged automatically at the end of the month, but that is not working anymore.

Maybe I am doing wrong but I don't get the issue, in that case, it would be great if you can explain it to me :smile: . Otherwise, would  be great to add a test with the flow in order to ensure that all works fine (it takes more than 10 minutes to test it manually, with a test, it takes seconds :wink: )

**Comment by kaynnan** [comment]
> Still not working. I am doing the following:
> 
>     * Create 2 companies  and make one of them consolidated
> 
>     * Add some movements for previous month on subsidiary
> 
>     * Match all the accounts (I use the same chart of accounts)
> 
>     * Consolidate previous month
> 
>     * A move is created
> 
>     * Consolidate previous month again
> 
>     * A move is created (should modify the old one or delete it)
> 
> 
> Also, previously the moves were marked to be merged automatically at the end of the month, but that is not working anymore.
> 
> Maybe I am doing wrong but I don't get the issue, in that case, it would be great if you can explain it to me 😄 . Otherwise, would be great to add a test with the flow in order to ensure that all works fine (it takes more than 10 minutes to test it manually, with a test, it takes seconds 😉 )

I'll review the situation and try to better understand the problem. Regarding the automatic merging action at the end of the month, I recall that functionality, but I'll investigate further. Thanks for providing the detailed steps, and I agree that creating an automated test could be an effective approach to expedite the flow verification. I'll follow the order you mentioned while conducting tests to identify and resolve the issue. Thanks for the review.

**Comment by kaynnan** [comment]
@etobella after reviewing the code, I made the necessary adjustments. My reference point was following the Runboat from v12. Below is the step-by-step process I followed. [Account Consolidation Runboat v12.0](https://runboat.odoo-community.org/webui/builds.html?repo=OCA/account-consolidation&target_branch=12.0).

1. Login to user (consolidation):

- User: consolidation
- Password: consolidation

2. Switch to Sub A and create Journal Entry:

image

3. Switch to Sub B and create Journal Entry:

image


4. Switch to Consolidation Company and run Consolidation Wizard:

image

5. Result Consolidation Items:

image
image

6. Run Consolidation Wizard Again (For Reversal Action):

image

7. Compare with the method in v12:

image

**Comment by kaynnan** [comment]
@etobella after reviewing the code, I made the necessary adjustments. My reference point was following the [Account Consolidation Runboat v12.0](https://runboat.odoo-community.org/webui/builds.html?repo=OCA/account-consolidation&target_branch=12.0). Below is the step-by-step process I followed. 

1. Login to user (consolidation):

- User: consolidation
- Password: consolidation

2. Switch to Sub A and create Journal Entry:

image

3. Switch to Sub B and create Journal Entry:

image


4. Switch to Consolidation Company and run Consolidation Wizard:

image

5. Result Consolidation Items:

image
image

6. Run Consolidation Wizard Again (For Reversal Action):

image

7. Compare with the method in v12:

image

**Comment by kaynnan** [comment]
> Still not working. I am doing the following:
> 
>     * Create 2 companies  and make one of them consolidated
> 
>     * Add some movements for previous month on subsidiary
> 
>     * Match all the accounts (I use the same chart of accounts)
> 
>     * Consolidate previous month
> 
>     * A move is created
> 
>     * Consolidate previous month again
> 
>     * A move is created (should modify the old one or delete it)
> 
> 
> Also, previously the moves were marked to be merged automatically at the end of the month, but that is not working anymore.
> 
> Maybe I am doing wrong but I don't get the issue, in that case, it would be great if you can explain it to me 😄 . Otherwise, would be great to add a test with the flow in order to ensure that all works fine (it takes more than 10 minutes to test it manually, with a test, it takes seconds 😉 )

@etobella after reviewing the code, I made the necessary adjustments. My reference point was following the [Account Consolidation Runboat v12.0](https://runboat.odoo-community.org/webui/builds.html?repo=OCA/account-consolidation&target_branch=12.0). Below is the step-by-step process I followed. 

1. Login to user (consolidation):

- User: consolidation
- Password: consolidation

2. Switch to Sub A and create Journal Entry:

image

3. Switch to Sub B and create Journal Entry:

image


4. Switch to Consolidation Company and run Consolidation Wizard:

image

5. Result Consolidation Items:

image
image

6. Run Consolidation Wizard Again (For Reversal Action):

image

7. Compare with the method in v12:

image

**Comment by kaynnan** [comment]
ping

I performed additional validations, and in this latest commit, it already includes the correction regarding the automatic posting at the end of the month. Previously, it was controlled by the `auto_reverse` flag, and a date was selected for reversal. Now, it has become `auto_post`, where the posting happens on the last day of the month.

**Review by WesleyOliveira98 — APPROVED** [review]

**Review by marcelsavegnago — APPROVED** [review]

**Comment by kaynnan** [comment]
ping @OCA/accounting-maintainers

**Pull request merged**